src/state/routing/v2Slice.ts

```diff
@@ -0,0 +1,120 @@
+  ...CLIENT_PARAMS,
+  routingType: 'CLASSIC',
+}
+
+export const routingApiV2 = createApi({
+  reducerPath: 'routingApiV2',
+  baseQuery: fetchBaseQuery({
+    baseUrl: 'https://api.uniswap.org/v2/',
+  }),
+  endpoints: (build) => ({
+    getQuote: build.query<TradeResult, GetQuoteArgs>({
+      async onQueryStarted(args: GetQuoteArgs, { queryFulfilled }) {
+        trace(
+          'quote-v2',
+          async ({ setTraceError, setTraceStatus }) => {
+            try {
+              await queryFulfilled
+            } catch (error: unknown) {
+              if (error && typeof error === 'object' && 'error' in error) {
+                const queryError = (error as Record<'error', FetchBaseQueryError>).error
+                if (typeof queryError.status === 'number') {
+                  setTraceStatus(queryError.status)
+                }
+                setTraceError(queryError)
+              } else {
+                throw error
+              }
+            }
+          },
+          {
+            data: {
+              ...args,
+              isPrice: args.routerPreference === INTERNAL_ROUTER_PREFERENCE_PRICE,
+              isAutoRouter:
+                args.routerPreference === RouterPreference.AUTO || args.routerPreference === RouterPreference.API,
+            },
+          }
+        )
+      },
+      async queryFn(args: GetQuoteArgs, _api, _extraOptions, fetch) {
+        const routerPreference = args.routerPreference
+        if (shouldUseAPIRouter(routerPreference)) {
+          try {
+            const { tokenInAddress, tokenInChainId, tokenOutAddress, tokenOutChainId, amount, tradeType } = args
+            const type = isExactInput(tradeType) ? 'EXACT_INPUT' : 'EXACT_OUTPUT'
+
+            const requestBody = {
+              tokenInChainId,
+              tokenIn: tokenInAddress,
+              tokenOutChainId,
+              tokenOut: tokenOutAddress,
+              amount,
+              type,
+              configs: [CLASSIC_SWAP_QUERY_PARAMS],
+            }
+
+            const response = await fetch({
+              method: 'POST',
+              url: '/quote',
+              body: JSON.stringify(requestBody),
+            })
+
+            if (response.error) {
+              try {
+                // cast as any here because we do a runtime check on it being an object before indexing into .errorCode
+                const errorData = response.error.data as any
+                // NO_ROUTE should be treated as a valid response to prevent retries.
+                if (typeof errorData === 'object' && errorData?.errorCode === 'NO_ROUTE') {
+                  return { data: { state: QuoteState.NOT_FOUND } }
+                }
+              } catch {
+                throw response.error
+              }
+            }
+
+            const quoteData = response.data as QuoteDataV2
+            const tradeResult = transformRoutesToTrade(args, quoteData.quote)
+
+            return { data: tradeResult }
```

Comment:
these lines are new logic - note v2 quote requests is a POST instead of GET, and has slightly different response/error formats